### PR TITLE
all: Rename occurances of "Apply" to "Save", fix racy TestNetworkingBasic pixel test

### PR DESCRIPTION
**Pull request by garrett**
"Apply" buttons should be "Save".

Reason: "Save" always means "make the changes and close this dialog", whereas "Apply" sometimes does this too but more often means "Just make the changes now and keep the dialog open", famously in Windows, where most of our users are. Making the options "Save" and "Cancel" is much clearer.

### CODE REVIEW

**Comment by garrett** [comment]
This probably changes a little too much (most likely in the storage page and perhaps in the software updates page). I wanted to share it as-is as a draft to let the tests run and hit the existing pixel testing to see what happens.

Meanwhile, I'll manually review all the changes too.

**Comment by garrett** [comment]
> Hmm, no strong option here. In these dialogs you make changes and then want to apply them. To me Apply seems like better word for that kind of change, but really cannot say if Save is better for user man_shrugging

Consistency is good. It's saving the changes.

But, yeah, this whole PR is kind of low priority and there might be unintended string changes.

**Comment by mvollmer** [comment]
For Storage:
- "Save" for "Stored Pasphrase" makes sense, it just writes to a file in /etc/luks-leys.
- "Save" for "Encryption Options" makes sense, it just writes to /etc/crypttab.
- "Save" for the "Mount" dialog is not so clear to me. It will also unmount and re-mount the filesystem. Do we want to include changes to the actual run-time in "Save"?

In general, I would rather follow GNOME conventions than Windows conventions.

GNOME Disks has a "Ok" button when changing fstab, and a "Change" button when changing the filesystem label.  However, GNOME Disks doesn't feel very polished or up-to-date these days.

GNOME Settings uses "Apply" when changing network settings.

But then again, I couldn't find anything about dialogs in the GNOME HIG...

**Comment by garrett** [comment]
> In general, I would rather follow GNOME conventions than Windows conventions.

This is about Cockpit and PatternFly conventions, not GNOME or Windows. That said, using Save is more like GNOME; Apply is more like Windows.

https://www.howtogeek.com/howto/15185/hey-you-stop-using-the-apply-button-and-just-click-ok-geek-rants/

"Apply" does not close a dialog, especially on Windows, whereas "Save" does, which is the point of renaming the buttons in this PR... as the actions we have in modals all intentionally close the dialog in Cockpit.

As a large amount of our users actually come from Windows, we should respect that on Windows, "Apply" does not close dialogs, so therefore, it should not be used on modals when actions would close the dialogs.

> But then again, I couldn't find anything about dialogs in the GNOME HIG...

GNOME usually uses Save or Update, not Apply. It's not called out explicitly in the HIG, but it is hinted at:
https://developer.gnome.org/hig/patterns/controls/buttons.html?highlight=save

---

As far as PatternFly is concerned, it has "Save" in the terminology (with its counterpart opposite of "Cancel" mentioned), but does not even define "Apply".

https://www.patternfly.org/v4/ux-writing/terminology/

---

> "Save" for the "Mount" dialog is not so clear to me. It will also unmount and re-mount the filesystem. Do we want to include changes to the actual run-time configuration in "Save"?

We could use "Update" (which is a bit generic and might be odd, given the context of the rest of the modals having save) or "Save and remount" (which is specific and consistent using the word "save", with a second action being included in the button).

At first, I thought "perhaps we should hyphenate re-mount", but it's a word in the MW dictionary. https://www.merriam-webster.com/dictionary/remount ... and it means:

> re·​mount | \ ˈrē-ˌmau̇nt \\ (noun)
> : a fresh horse to replace one no longer available

But it also means "to mount (something) again", which is more appropriate for Cockpit. 🏇

**Comment by martinpitt** [comment]
> Reason: "Save" always means "make the changes and close this dialog", whereas "Apply" sometimes does this too but more often means "Just make the changes now and keep the dialog open

That's not really part of the definition of the words, right? Just common practice from Windows? E.g. "Save" in Word doesn't close the document either.

**Comment by garrett** [comment]
> That's not really part of the definition of the words, right? Just common practice from Windows? E.g. "Save" in Word doesn't close the document either.

Correct, but you have to think about context and what people using Cockpit (and especially Windows) expect.

Save on the save document modal dialog (common save dialog) in Word _closes the save dialog window_.

Apply on modals in Windows do not close the dialogs, but save always does.

And this isn't just about people who use Windows either. On Linux, KDE is rather popular, and they use the same OK / Apply / Cancel paradigm as Windows. (Where "apply" does not close the modal dialog.)

Here's a screenshot I just took from the latest version of KDE, with the file manager preferences dialog showing OK (which applies and closes the window), Apply (which only applies and doesn't close the window), and Cancel (which closes the window without applying). Just like on Windows, Apply is redundant as OK applies and closes the window. 

Screenshot from 2021-11-08 13-39-49

Clicking on Apply and Cancel would actually do the same thing as clicking on OK. 

I've actually seen lots of people in usability studies in the past as well as just watching people use Windows and KDE over the years click the Apply button first and then OK button. Either you're going to apply the settings or not and then you're going to close the window. All of those tasks are reflected in the OK (or Save or any other action) and Cancel buttons.

But the point I'm making here is that it's common for Windows and KDE users to expect an Apply button in modal dialogs to not close the window, therefore, it's a loaded term and if we close modals with the "Apply" button, it will diverge from their expectations. Hence this PR.

**Comment by garrett** [comment]
Windows has done a lot to get rid of the Apply button in Settings (as it's all in an instant-apply app now), but it still exists when you get to modal dialogs (especially older, untouched ones, which still show up in Windows 10 and 11). 

Here's a device dialog in Windows 10:

Screenshot from 2021-11-08 13-47-38

Apply doesn't close the dialog (and is disabled until changes are made, just like in KDE), but the other buttons (in this case OK and Cancel) do.

I'm assuming that Microsoft saw the same thing I had over the years and realized that Apply was an anti-pattern and tried to move a lot of their most common settings away from it in the Windows 8 & 10 era. (But it still even persists in Windows 11, as Microsoft never truly fully changes their UI, they just bury the old and rework the most common user-facing stuff.)

**Comment by martinpitt** [comment]
@garrett : Thanks! Sorry for the misunderstanding -- I didn't want to say "I disagree", just wanted to confirm that I understood it in the right way, as I haven't used Windows nor KDE in a long time.

In the interest of easing translator work, we should of course replace all buttons with just :+1: and :x: ! *cough*

**Comment by garrett** [comment]
Rebased

**Comment by garrett** [comment]
Rebased; also added network and kdump dialogs to the apply → save change.

**Comment by martinpitt** [comment]
I adjusted the tests for the renaming and [updated the pixel tests](https://github.com/cockpit-project/pixel-test-reference/compare/e37891c2b526d1de366e25ae4f57ef314990a45c..6579d45fef970177293dc1c666530fb150476109).

**Comment by martinpitt** [comment]
That [failure happened](https://logs.cockpit-project.org/logs/pull-16475-20220318-112949-5b57a2a4-fedora-35/log.html#61-1) because tests run multiple times. I added another commit which fixes that, and at the same time makes the pixel test a lot simpler.

As I now meddled too much with this, @marusak, can you please do the final review?

**Comment by martinpitt** [comment]
[Some](https://logs.cockpit-project.org/logs/pull-16475-20220318-172950-02d7d82c-arch/log.html#265) [unrelated](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-16475-20220318-172950-02d7d82c-debian-stable/log.html) flakes, I sent PR #17152 to fix them.

**Comment by martinpitt** [comment]
Ready for review. Can we please land this soon, as it is rather invasive? (pixel tests, lots of changed places)

File: pkg/kdump/kdump-view.jsx [hunk 1]
```diff
@@ -357,7 +357,7 @@ export class KdumpPage extends React.Component {
             actions: [
                 {
                     clicked: this.handleApplyClick.bind(this),
```
Comment (marusak) [inline pkg/kdump/kdump-view.jsx:359]: So this leaves behind `handleApplyClick`. In `pkg/networkmanager/dialogs-common.jsx` it leaves behind `id=...-apply`. These are just names so does not matter, just maybe would be a bit cleaner for future ourselves when we wonder why we call it "apply" :) `git grep "[aA]pply" pkg/` shows a lot of such cases. Not necessary to do anything, just pointing it out.

**Review by marusak — COMMENTED** [review]
There is one new occurrence in `pkg/systemd/overview-cards/cryptoPolicies.jsx` - It has `Apply and reboot` or `Apply only` buttons. I wonder if these should not also be `Save and reboot`/`Save only` :thinking:

File: pkg/kdump/kdump-view.jsx (hunk 1 above)
Comment (martinpitt) [reply pkg/kdump/kdump-view.jsx]: Good point! I renamed these, at least all that I could find which immediately make sense.

I also adjusted cryptoPolicies, that landed recently and we didn't have it on the radar in this PR.

**Review by marusak — APPROVED** [review]
Thank you!

**Pull request merged**